File: app/src/main/java/me/imzack/app/end/injector/component/PlanSearchComponent.java
package me.imzack.app.end.injector.component;

import me.imzack.app.end.injector.module.PlanSearchPresenterModule;
import me.imzack.app.end.injector.scope.ActivityScope;
import me.imzack.app.end.view.activity.PlanSearchActivity;

import dagger.Component;

@ActivityScope
@Component(dependencies = AppComponent.class, modules = PlanSearchPresenterModule.class)
public interface PlanSearchComponent {

    void inject(PlanSearchActivity planSearchActivity);
}

File: app/src/main/kotlin/me/imzack/app/end/injector/component/PlanSearchComponent.kt
package me.imzack.app.end.injector.component

import dagger.Component
import me.imzack.app.end.injector.module.PlanSearchPresenterModule
import me.imzack.app.end.injector.scope.ActivityScope
import me.imzack.app.end.view.activity.PlanSearchActivity

@ActivityScope
@Component(dependencies = arrayOf(AppComponent::class), modules = arrayOf(PlanSearchPresenterModule::class))
interface PlanSearchComponent {

    fun inject(planSearchActivity: PlanSearchActivity)
}
